File: internal/integrations/strava/sport_types.go
package strava

import (
	"strings"

	"github.com/solargate/grom/internal/workouts"
)

var sportTypeAliases = map[string]string{
	// English
	"run": "Run",
	"ride": "Ride",
	"walk": "Walk",
	"nordic walk": "NordicWalk",
	"nordic walking": "NordicWalk",
	"hike": "Hike",
	"swim": "Swim",
	"workout": "Workout",
	"weight training": "WeightTraining",
	"virtual ride": "Ride",
	"virtual run": "Run",
	"ebikeride": "EBikeRide",
	"e-bike ride": "EBikeRide",
	"emountainbikeride": "EMountainBikeRide",
	"mountain bike ride": "MountainBikeRide",
	"gravel ride": "GravelRide",
	"trail run": "TrailRun",
	"canoeing": "Canoeing",
	"kayaking": "Kayaking",
	"packraft": "Packraft",
	"stand up paddling": "SUP",
	"stand-up paddling": "SUP",
	"surfing": "Surfing",
	"kitesurf": "Kitesurf",
	"rowing": "Rowing",
	"windsurf": "Windsurf",
	"sail": "Sail",
	"ice skate": "IceSkate",
	"nordic ski": "NordicSki",
	"alpine ski": "AlpineSki",
	"snowboard": "Snowboard",
	"backcountry ski": "BackcountrySki",
	"ice hockey": "IceHockey",
	"hockey": "IceHockey",
	"snowshoe": "Snowshoe",
	"golf": "Golf",
	"badminton": "Badminton",
	"elliptical": "Elliptical",
	"basketball": "Basketball",
	"inline skate": "InlineSkate",
	"skateboard": "Skateboard",
	"tennis": "Tennis",
	"stair stepper": "StairStepper",
	"padel": "Padel",
	"rock climbing": "RockClimbing",
	"soccer": "Soccer",
	"pickleball": "Pickleball",
	"volleyball": "Volleyball",
	"roller ski": "RollerSki",
	"squash": "Squash",
	"crossfit": "Crossfit",
	"yoga": "Yoga",
	"dance": "Dance",
	"table tennis": "TableTennis",
	"pilates": "Pilates",
	"racquetball": "Racquetball",
	"hiit": "HIIT",
	"high intensity interval training": "HIIT",
	"highintensityintervaltraining": "HIIT",
	"cricket": "Cricket",
	"wheelchair": "Wheelchair",
	"handcycle": "Handcycle",
	"velomobile": "Velomobile",
	"water sport": "Workout",

	// Russian
	"бег": "Run",
	"велосипед": "Ride",
	"ходьба": "Walk",
	"скандинавская ходьба": "NordicWalk",
	"поход": "Hike",
	"плавание": "Swim",
	"тренировка": "Workout",
	"силовая тренировка": "WeightTraining",
	"виртуальный заезд": "Ride",
	"виртуальный бег": "Run",
	"заезд на электровелосипеде": "EBikeRide",
	"горный велозаезд": "MountainBikeRide",
	"гравийный велозаезд": "GravelRide",
	"трейлран": "TrailRun",
	"канoeing": "Canoeing",
	"каякинг": "Kayaking",
	"пакрафт": "Packraft",
	"гребля на сапе": "SUP",
	"серфинг": "Surfing",
	"кайтсерфинг": "Kitesurf",
	"гребля": "Rowing",
	"виндсерфинг": "Windsurf",
	"парусный спорт": "Sail",
	"катание на коньках": "IceSkate",
	"лыжи классические": "NordicSki",
	"горные лыжи": "AlpineSki",
	"сноуборд": "Snowboard",
	"скитур": "BackcountrySki",
	"хоккей": "IceHockey",
	"снегоступы": "Snowshoe",
	"гольф": "Golf",
	"бадминтон": "Badminton",
	"эллипс": "Elliptical",
	"баскетбол": "Basketball",
	"ролики": "InlineSkate",
	"скейтборд": "Skateboard",
	"теннис": "Tennis",
	"степпер": "StairStepper",
	"скалолазание": "RockClimbing",
	"футбол": "Soccer",
	"волейбол": "Volleyball",
	"сквош": "Squash",
	"кроссфит": "Crossfit",
	"йога": "Yoga",
	"танцы": "Dance",
	"настольный теннис": "TableTennis",
	"пилатес": "Pilates",
	"крикет": "Cricket",
	"инвалидная коляска": "Wheelchair",
	"ручной велосипед": "Handcycle",
	"веломобиль": "Velomobile",
	"силовая": "WeightTraining",
}

var nameSportKeywords = []struct {
	keyword string
	sport   string
}{
	{"пакрафт", "Packraft"},
	{"packraft", "Packraft"},
	{"пилатес", "Pilates"},
	{"pilates", "Pilates"},
	{"йога", "Yoga"},
	{"yoga", "Yoga"},
	{"кроссфит", "Crossfit"},
	{"crossfit", "Crossfit"},
	{"hiit", "HIIT"},
	{"растяжка", "Yoga"},
	{"stretching", "Yoga"},
	{"танцы", "Dance"},
	{"dance", "Dance"},
	{"эллипс", "Elliptical"},
	{"elliptical", "Elliptical"},
	{"степпер", "StairStepper"},
	{"stair stepper", "StairStepper"},
}

func normalizeSportKey(raw string) string {
	raw = strings.ToLower(strings.TrimSpace(raw))
	raw = strings.ReplaceAll(raw, "\u00a0", " ")
	raw = strings.ReplaceAll(raw, "\u202f", " ")
	return strings.Join(strings.Fields(raw), " ")
}

func inferSportFromName(name string) string {
	key := normalizeSportKey(name)
	if key == "" {
		return ""
	}
	for _, item := range nameSportKeywords {
		if strings.Contains(key, item.keyword) {
			return item.sport
		}
	}
	return ""
}

func mapSportType(raw, name string) (string, error) {
	key := normalizeSportKey(raw)
	if key == "" {
		return "", errInvalidSport
	}
	if mapped, ok := sportTypeAliases[key]; ok {
		if mapped == "Workout" {
			if inferred := inferSportFromName(name); inferred != "" {
				return inferred, nil
			}
		}
		return mapped, nil
	}
	// Direct grom sport type passthrough.
	candidate := strings.ReplaceAll(key, " ", "")
	for _, sport := range workouts.ValidSportTypes() {
		if strings.EqualFold(sport, raw) || strings.EqualFold(sport, candidate) {
			return sport, nil
		}
	}
	if inferred := inferSportFromName(name); inferred != "" {
		return inferred, nil
	}
	return "Workout", nil
}

File: ui/grom/lib/models/sport_types.dart
import 'package:flutter/material.dart';
import 'package:grom/l10n/app_localizations.dart';
import 'package:grom/l10n/sport_type_localizations.dart';
import 'package:material_symbols_icons/symbols.dart';

enum SportCategory {
  foot,
  cycle,
  strength,
  water,
  winter,
  team,
  racket,
  other,
}

class SportTypeInfo {
  const SportTypeInfo({
    required this.id,
    required this.category,
    required this.icon,
  });

  final String id;
  final SportCategory category;
  final IconData icon;
}

const defaultSportTypeId = 'Run';

const sportTypeCatalog = <SportTypeInfo>[
  // foot
  SportTypeInfo(id: 'Run', category: SportCategory.foot, icon: Icons.directions_run),
  SportTypeInfo(id: 'Walk', category: SportCategory.foot, icon: Icons.directions_walk),
  SportTypeInfo(id: 'NordicWalk', category: SportCategory.foot, icon: Icons.nordic_walking),
  SportTypeInfo(id: 'Hike', category: SportCategory.foot, icon: Icons.hiking),
  SportTypeInfo(id: 'TrailRun', category: SportCategory.foot, icon: Icons.directions_run),
  SportTypeInfo(id: 'Wheelchair', category: SportCategory.foot, icon: Icons.accessible),
  // cycle
  SportTypeInfo(id: 'Ride', category: SportCategory.cycle, icon: Icons.directions_bike),
  SportTypeInfo(id: 'EBikeRide', category: SportCategory.cycle, icon: Icons.electric_bike),
  SportTypeInfo(id: 'MountainBikeRide', category: SportCategory.cycle, icon: Icons.pedal_bike),
  SportTypeInfo(id: 'EMountainBikeRide', category: SportCategory.cycle, icon: Icons.electric_bolt),
  SportTypeInfo(id: 'GravelRide', category: SportCategory.cycle, icon: Icons.directions_bike),
  SportTypeInfo(id: 'Velomobile', category: SportCategory.cycle, icon: Icons.sports_motorsports),
  SportTypeInfo(id: 'Handcycle', category: SportCategory.cycle, icon: Icons.two_wheeler),
  // strength
  SportTypeInfo(id: 'Workout', category: SportCategory.strength, icon: Symbols.physical_therapy),
  SportTypeInfo(id: 'WeightTraining', category: SportCategory.strength, icon: Symbols.exercise),
  SportTypeInfo(id: 'HIIT', category: SportCategory.strength, icon: Icons.local_fire_department),
  SportTypeInfo(id: 'Crossfit', category: SportCategory.strength, icon: Icons.fitness_center),
  // water
  SportTypeInfo(id: 'Canoeing', category: SportCategory.water, icon: Icons.kayaking),
  SportTypeInfo(id: 'SUP', category: SportCategory.water, icon: Icons.surfing),
  SportTypeInfo(id: 'Kayaking', category: SportCategory.water, icon: Icons.kayaking),
  SportTypeInfo(id: 'Packraft', category: SportCategory.water, icon: Icons.kayaking),
  SportTypeInfo(id: 'Surfing', category: SportCategory.water, icon: Icons.surfing),
  SportTypeInfo(id: 'Kitesurf', category: SportCategory.water, icon: Icons.kitesurfing),
  SportTypeInfo(id: 'Swim', category: SportCategory.water, icon: Icons.pool),
  SportTypeInfo(id: 'Rowing', category: SportCategory.water, icon: Icons.rowing),
  SportTypeInfo(id: 'Windsurf', category: SportCategory.water, icon: Icons.sailing),
  SportTypeInfo(id: 'Sail', category: SportCategory.water, icon: Icons.sailing),
  // winter
  SportTypeInfo(id: 'IceSkate', category: SportCategory.winter, icon: Icons.ice_skating),
  SportTypeInfo(id: 'NordicSki', category: SportCategory.winter, icon: Icons.downhill_skiing),
  SportTypeInfo(id: 'AlpineSki', category: SportCategory.winter, icon: Icons.downhill_skiing),
  SportTypeInfo(id: 'Snowboard', category: SportCategory.winter, icon: Icons.snowboarding),
  SportTypeInfo(id: 'BackcountrySki', category: SportCategory.winter, icon: Icons.downhill_skiing),
  SportTypeInfo(id: 'IceHockey', category: SportCategory.winter, icon: Icons.sports_hockey),
  SportTypeInfo(id: 'Snowshoe', category: SportCategory.winter, icon: Icons.snowshoeing),
  // team
  SportTypeInfo(id: 'Soccer', category: SportCategory.team, icon: Icons.sports_soccer),
  SportTypeInfo(id: 'Basketball', category: SportCategory.team, icon: Icons.sports_basketball),
  SportTypeInfo(id: 'Volleyball', category: SportCategory.team, icon: Icons.sports_volleyball),
  SportTypeInfo(id: 'Cricket', category: SportCategory.team, icon: Icons.sports_cricket),
  // racket
  SportTypeInfo(id: 'Tennis', category: SportCategory.racket, icon: Icons.sports_tennis),
  SportTypeInfo(id: 'Padel', category: SportCategory.racket, icon: Icons.sports_tennis),
  SportTypeInfo(id: 'Pickleball', category: SportCategory.racket, icon: Symbols.pickleball),
  SportTypeInfo(id: 'Racquetball', category: SportCategory.racket, icon: Icons.sports_tennis),
  SportTypeInfo(id: 'Squash', category: SportCategory.racket, icon: Icons.sports_tennis),
  SportTypeInfo(id: 'Badminton', category: SportCategory.racket, icon: Symbols.badminton),
  SportTypeInfo(id: 'TableTennis', category: SportCategory.racket, icon: Icons.sports_tennis),
  // other
  SportTypeInfo(id: 'Golf', category: SportCategory.other, icon: Icons.golf_course),
  SportTypeInfo(id: 'Elliptical', category: SportCategory.other, icon: Icons.monitor_heart),
  SportTypeInfo(id: 'InlineSkate', category: SportCategory.other, icon: Icons.roller_skating),
  SportTypeInfo(id: 'Skateboard', category: SportCategory.other, icon: Icons.skateboarding),
  SportTypeInfo(id: 'StairStepper', category: SportCategory.other, icon: Icons.stairs),
  SportTypeInfo(id: 'RockClimbing', category: SportCategory.other, icon: Icons.landscape),
  SportTypeInfo(id: 'RollerSki', category: SportCategory.other, icon: Icons.nordic_walking),
  SportTypeInfo(id: 'Yoga', category: SportCategory.other, icon: Icons.self_improvement),
  SportTypeInfo(id: 'Dance', category: SportCategory.other, icon: Icons.music_note),
  SportTypeInfo(id: 'Pilates', category: SportCategory.other, icon: Icons.self_improvement),
];

SportTypeInfo? sportTypeById(String id) {
  for (final sportType in sportTypeCatalog) {
    if (sportType.id == id) {
      return sportType;
    }
  }
  return null;
}

String sportCategoryLabel(AppLocalizations l10n, SportCategory category) {
  switch (category) {
    case SportCategory.foot:
      return l10n.sportCategoryFoot;
    case SportCategory.cycle:
      return l10n.sportCategoryCycle;
    case SportCategory.strength:
      return l10n.sportCategoryStrength;
    case SportCategory.water:
      return l10n.sportCategoryWater;
    case SportCategory.winter:
      return l10n.sportCategoryWinter;
    case SportCategory.team:
      return l10n.sportCategoryTeam;
    case SportCategory.racket:
      return l10n.sportCategoryRacket;
    case SportCategory.other:
      return l10n.sportCategoryOther;
  }
}

String sportTypeLabel(AppLocalizations l10n, String id) {
  return sportTypeLocalization(l10n, id);
}

Color sportTypeColor(String id) {
  final info = sportTypeById(id);
  if (info == null) {
    return Colors.grey;
  }
  switch (info.category) {
    case SportCategory.foot:
      return const Color(0xFFFC4C02);
    case SportCategory.cycle:
      return const Color(0xFF1E88E5);
    case SportCategory.strength:
      return const Color(0xFF8A9440);
    case SportCategory.water:
      return const Color(0xFF00897B);
    case SportCategory.winter:
      return const Color(0xFF5C6BC0);
    case SportCategory.team:
      return const Color(0xFFC45C5C);
    case SportCategory.racket:
      return const Color(0xFF5E8A5E);
    case SportCategory.other:
      return const Color(0xFF8E24AA);
  }
}
